README.md

```diff
@@ -18,6 +28,87 @@ Then build the backup tools.
 
 This gives you two binaries in the `bin` subdirectory -- `asbackup` and `asrestore`.
 
+## Build Examples
+
+Debian and Ubuntu
+
+    apt-get update
+
+    # Install C client dependencies...
+
+    # asbackup dependencies
+    apt-get install build-essential libssl-dev libuv1-dev libcurl4-openssl-dev libzstd-dev
+
+    # for aws-sdk-cpp build
+    apt-get install cmake
+
+    # download aws sdk
+    git clone https://github.com/aws/aws-sdk-cpp.git
+    cd aws-sdk-cpp
+    git submodule update --init --recursive
+
+    # build aws sdk dynamic
+    mkdir build
+    cmake -S . -B build -DCMAKE_BUILD_TYPE=Release -DBUILD_ONLY="s3" -DBUILD_SHARED_LIBS=ON -DENABLE_TESTING=OFF -DCMAKE_INSTALL_PREFIX=/usr/local -DCMAKE_INSTALL_LIBDIR=lib
+    make -C build
+
+    # install aws static sdk
+    cd build
+    make install
+    cd ../..
```

Comment:
I think all these examples should use static linking for aws-sdk.  Maybe use static linking for it all?  I am not sure what the norm is here but as a user I would want everything statically linked except maybe openssl

Reply:
Maybe instead specify that these will dynamically link everything.  It would be nice to add a statically linked example too. This readme will likely be used by people on the tools team more than customers and we always statically link

Reply:
I figured users would normally be building from source and running the tool on the same machine so dynamic linking was fine (plus simpler). I'll add one static example and dynamic/static labels. I think the devs can reference the github actions over anything else.